<div class="content" role="main">
  <h1>Throwing and returning</h1>

  <div class="card card--headed">
    <div class="header">Some other details</div>
    <ul>
      <li><code>return</code> is used to set the value when <code>{{ '{' }} done: true }</code></li>
      <li>The return value, however, isn't included when iterating using spread, or for...of</li>
      <li>Throwing an exception inside a generator will return that error on <code>next()</code></li>
      <li>...and will complete the generator</li>
    </ul>

    <app-console>
      <pre>function* myGenerator() {{ '{' }}</pre>
      <pre>  throw new Error('Oops...');</pre>
      <pre>}</pre>
      <pre>

      </pre>
      <pre>const generator = myGenerator();</pre>
      <pre>

      </pre>

      <pre>generator.next(); <span class="comment">// Uncaught Error: Oops...</span></pre>
    </app-console>
  </div>

  <nav>
    <a routerLink="/examples/3" class="card card--prev card--mini">
      <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" fill="#1976d2"/><path d="M0 0h24v24H0z" fill="none"/></svg>
      <span>Yielding other iterables</span>
    </a>
    <a routerLink="/examples/5" class="card card--next card--mini">
      <span>Internal state</span>
      <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" fill="#1976d2"/><path d="M0 0h24v24H0z" fill="none"/></svg>
    </a>
  </nav>
</div>
